substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationManagementExt.java

```diff
@@ -77,7 +79,7 @@ private void linkManagementExt(@SuppressWarnings("unused") DuringAnalysisAccess
 
     @Override
     public void afterAnalysis(AfterAnalysisAccess access) {
-        if (NativeLibrarySupport.singleton().isPreregisteredBuiltinLibrary("awt_headless")) {
+        if (managementExtNeeded && NativeLibrarySupport.singleton().isPreregisteredBuiltinLibrary("awt_headless")) {
```

Comment:
Instead of using a field, we can't check somehow that "management_ext" is already added?

Reply:
OK. Done now. It requires iterating over the internal list of static libraries, though (using `NativeLibraries`).